app/code/Magento/ProductAlert/Block/Adminhtml/Catalog/Product/Edit/Tab/Alerts/Stock.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Magento\ProductAlert\Block\Adminhtml\Catalog\Product\Edit\Tab\Alerts;
+
+use Magento\Backend\Block\Widget\Grid\Extended;
+
+class Stock extends \Magento\Catalog\Block\Adminhtml\Product\Edit\Tab\Alerts\Stock
+{
+    /**
+     * @return $this
+     * @throws \Exception
+     */
+    protected function _prepareColumns()
+    {
+        $this->addColumn('firstname', ['header' => __('First Name'), 'index' => 'firstname']);
```

Comment:
Need to avoid duplication of structure
Because we can not reflect changes in parent class

Reply:
Ok. I totally agree about that. I don't like code duplication either.
 I'll try to do that another way.